agents/temporal_difference.c

```diff
@@ -0,0 +1,133 @@
+#include <float.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "game.h"
+#include "temporal_difference.h"
+#include "util.h"
+
+// TODO: Find a more efficient hash, we could not store 5x5 or larger board,
+// Since we could have 3^25 states and it might overflow.
+int table_to_hash(char *table)
+{
+    int ret = 0;
+    for (int i = 0; i < N_GRIDS; i++) {
+        ret *= 3;
+        if (table[i] == ' ') {
+            ret += 0;
+        } else if (table[i] == 'O') {
+            ret += 1;
+        } else if (table[i] == 'X') {
+            ret += 2;
+        }
+    }
+    return ret;
+}
+
+char *hash_to_table(int hash)
```

Comment:
Since we are not going to expose the function, we should add the 'static' modifier to protect the scope that can access the function.